workflow/controller/cache/configmap_cache.go

```diff
@@ -60,7 +61,12 @@ func (c *configMapCache) Load(ctx context.Context, key string) (*Entry, error) {
 	}
 
 	c.logInfo(log.Fields{}, "config map cache loaded")
-
+	cm.SetLabels(map[string]string{common.LabelKeyCacheLastHitTimestamp: time.Now().Format(time.RFC3339)})
```

Comment:
It seems like we'd want this to be per-entry?

Reply:
I thought about that too. However, when we need to delete the individual configmaps, all entries on that configmap would be deleted. Isn't this what you also suggested in https://github.com/argoproj/argo-workflows/issues/5406#issuecomment-802175117?

I do think it's also useful to see per entry hit timestamps though. Perhaps we can add both? What do you think?

Reply:
Actually I just updated the PR to include the last hit timestamps for each entry as well since we can do update all these information in one update call and these timestamps per entry and per cache would both be useful to decide a GC strategy.